### PR TITLE
Add EShop + Add SSB

### PR DESCRIPTION
**Pull request by zellman01**

### CODE REVIEW

File: chat-plugins/eShop.js [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+    for spacialgaze.psim.us
+    coded by HoeenHero
+    -----------------------
+    THIS IS NOT THE SAME AS
+    THE PRIVATE SHOP ON SG
+******************************/
+'use strict';
+
+const fs = require('fs');
+let allowThisShop = false; //Change to true to make these command work
```
Comment (DeathlyPlays) [inline chat-plugins/eShop.js]: Set this to true since we have no other shop

File: chat-plugins/eShop.js [hunk 2]
```diff
@@ -0,0 +1,230 @@
...
+	fs.accessSync('chat-plugins/shop.js', fs.F_OK);
+	if (allowThisShop) console.warn('Since the normal shop is up the eShop has been disabled.');
+	allowThisShop = false;
+} catch (e) {
+	if (!allowThisShop) console.warn('Unable to find the normal shop, activating the eShop...');
+	allowThisShop = true;
+}
+
+exports.commands = {
+	//shop: 'eshop', //Uncomment this if you want this to be able to be used using the /shop command
```
Comment (DeathlyPlays) [inline chat-plugins/eShop.js]: Uncomment this out

File: chat-plugins/eShop.js [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+		buy: function (target, room, user, connection, cmd, message) {
+			if (!allowThisShop) return this.errorReply('This shop is closed');
+			if (!target) return this.parse('/eshop help buy');
+			if (OCPU.eShop.closed) return this.sendReply('The shop is closed, come back later.');
+			if (!OCPU.eShop[toId(target)]) return this.errorReply('Item ' + target + ' not found.');
+			let item = OCPU.eShop[toId(target)];
+			Economy.readMoney(user.userid, userMoney => {
+				if (item.price > userMoney) return this.errorReply('You need ' + (item.price - userMoney) + ' more ' + ((item.price - userMoney) === 1 ? global.currencyName : global.currenyPlural) + ' to buy this.');
+				if (item.isSSB && !OCPU.ssb[user.userid]) return this.sendReply('You need to run /ssb edit at least once before you can buy this.');
+				/*if (item.isSSB) {
```
Comment (DeathlyPlays) [inline chat-plugins/eShop.js]: If you are going to comment this out, comment other things mentioning isSSB or delete it all; however we might be better off just adding ssbffa

Comment (zellman01) [reply chat-plugins/eShop.js]: It would be better if we just did add it

Comment (DeathlyPlays) [reply chat-plugins/eShop.js]: Yeah, SGs EShop heavily mentions SSBFFA lmfao

File: data/sgssb-data.txt [hunk 4]
```diff
@@ -0,0 +1,348 @@
+HoeenHero
```
Comment (DeathlyPlays) [inline data/sgssb-data.txt]: Why?

File: mods/ocpussb/abilities.js [hunk 5]
```diff
@@ -0,0 +1,265 @@
+'use strict';
```
Comment (DeathlyPlays) [inline mods/ocpussb/abilities.js]: Why

File: mods/ocpussb/items.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+'use strict';
```
Comment (DeathlyPlays) [inline mods/ocpussb/items.js]: Why?

File: data/sgssb-data.txt (hunk 4 above)
Comment (zellman01) [reply data/sgssb-data.txt]: By default what was there

File: mods/ocpussb/abilities.js (hunk 5 above)
Comment (zellman01) [reply mods/ocpussb/abilities.js]: By default what was there

File: mods/ocpussb/items.js (hunk 6 above)
Comment (zellman01) [reply mods/ocpussb/items.js]: By default what was there

Comment (DeathlyPlays) [reply mods/ocpussb/items.js]: But why have SG's SSB when this SSB is not necessary to exist

Comment (zellman01) [reply mods/ocpussb/items.js]: Might as well add it mostly because someone later down the line if we don't have it will probably ask for it to be added

**Pull request merged**